### ISSUE TITLE
Get inspiration from NBLI and a lossless image compression benchmark

### ISSUE THREAD

**Issue by WangXuan95** [nigeltao/qoi2-bikeshed#44]
I hope my work can help QOI2:
- [A lossless image compression benchmark](https://github.com/WangXuan95/Image-Compression-Benchmark), which tests various lossless image compression algorithms on several large image sets.
  - stardard algorithms: PNG, JPEG-XL, JPEG-LS, QOI, etc.
  - non-stardard algorithms: [LEA](https://encode.su/threads/3818-LEA-Lossless-image-compressor), [BMF](http://compression.ru/compression.ru/ds/), [FLIC](http://www.imagecompression.info/gralic/flic21d.zip), [NBLI](https://github.com/WangXuan95/NBLI), etc.
- [NBLI](https://github.com/WangXuan95/NBLI): A fast, quite good, semi-open-sourced lossless image compressor. The optimized version (fNBLI) can get 1.5x compression ratio, 8x faster encoding, and 1.2x faster decoding compared to PNG. It can also beat the SOTA JPEG-XL. Actually, the baseline NBLI is improved from the following techniques:
  - [CALIC lossless image compressor](https://ieeexplore.ieee.org/abstract/document/585919)
  - [Glicbawls lossless image compressor](https://www.researchgate.net/publication/221578041_Glicbawls_-_Grey_Level_Image_Compression_by_Adaptive_Weighted_Least_Squares)
  - [JPEG-LS extension](https://github.com/WangXuan95/JPEG-LS_extension) ([ITU-T T.870](https://www.itu.int/rec/T-REC-T.870/en))
  - [EDP: Edge-directed predictor](https://ieeexplore.ieee.org/abstract/document/923277)

**Comment by nigeltao** [comment]
The compression ratio numbers for NBLI look impressive but bear in mind that QOI (and any possible QOI2) prioritizes (1) simplicity of implementation and (2) compression and decompression speed over (3) compression ratio.

Is there a description of what the NBLI codec actually is? There's the source code, obviously. I also skimmed the README.md's Acknowledgements section and there is a bunch of prior art mentioned but I didn't see an overview of how NBLI works algorithmically.

**Comment by p0nce** [comment]
@WangXuan95 I tested it it's pretty cool, especially considering HARLIC has no source available.
It easily beats JPEG and PNG here, I was more after the lossy modes interesting considering the reduced complexity.
Interestingly it looks better because of some kind of noise modelling? It had more HF.
I see images encoded with NBLI are not decoded with fNBLI, is that a different bitstream?

tl;dr you should mention it is lossy as well!
